Create innovo-inventor.json

innovo3d opened the pull request:
This is a custom machine profile for the Innovo Inventor.  More info at www.innovo3d.co.  Please feel free to write to me if you have any comments at adam@innovo3d.co

BagelOrb commented:
Please look at the overrides section as it is in other json files. Your bed temp visibility should be an override, independent of the category structure.

Change that and we'll merge your PR.

adamarumjahn commented on resources/settings/innovo-inventor.json:
```diff
@@ -0,0 +1,49 @@
+{
+  "id": "innovo-inventor",
+  "name": "Innovo INVENTOR",
+  "icon": "icon_ultimaker.png",
```
How do I add my own icon/logo? Where can I include the png file?

adamarumjahn commented on resources/settings/innovo-inventor.json:
```diff
@@ -0,0 +1,49 @@
+  "id": "innovo-inventor",
+  "name": "Innovo INVENTOR",
+  "icon": "icon_ultimaker.png",
+  "platform": "",
```
What's this? Is it to load an STL file on the platform when you start?

nallath commented:
The icon isn't used in the current visual design.

BagelOrb commented:
Please look at the other json files for answers on your questions. Try opening ultimaker_platform.stl and see what it is.

adamarumjahn commented:
Ok.  Thanks Tim.

Is there anything else that you need fixed before merging?  Please let me
know.  Thanks.

*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Fri, Aug 28, 2015 at 4:26 AM, Tim Kuipers <notifications@github.com>
wrote:

> Please look at the other json files for answers on your questions. Try
> opening ultimaker_platform.stl and see what it is.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-135710049>.
>

BagelOrb commented:
It's almost totally up to standard.
Please make it so that the indents 'tabs' are 4 spaces - not two.

adamarumjahn commented:
Hi Tim,

Fixed the spaces in the indents to 4.  I've also added a referene to inventor_platform.stl.

adamarumjahn commented:
Tim, how do I add my inventor_platform.stl file to the resources/meshes folder?  I don't seem to have permission to add a file.  Sorry, I'm still a n00b to Github

BagelOrb commented:
The JSON file is invalid. Please fix it. Verify it at http://jsonlint.com/

The machine_start_gcode is missing a quotation mark.

Just like you made a pull request to add innovo-inventor.json, you should also make a pull request for the inventor_platform.stl.

Please view https://github.com/Ultimaker/Cura/issues/243 to make sure your platform mesh is up to standard.

adamarumjahn commented:
Thanks Tim.  I fixed the JSON file.  I also decreased the STL file down
from 13MB to 811kb.  Please let me know if that works and if you need
anything else.

*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Tue, Sep 8, 2015 at 3:18 AM, Tim Kuipers <notifications@github.com>
wrote:

> The JSON file is invalid. Please fix it. Verify it at http://jsonlint.com/
>
> The machine_start_gcode is missing a quotation mark.
>
> Just like you made a pull request to add innovo-inventor.json, you should
> also make a pull request for the inventor_platform.stl.
>
> Please view #243 <https://github.com/Ultimaker/Cura/issues/243> to make
> sure your platform mesh is up to standard.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-138472625>.
>

BagelOrb commented:
It's best if you add the stl in this pull request.

adamarumjahn commented:
Hi Tim,

Please forgive me for asking.  I'm still quite new to Github.

How do I add the STL file in this pull request?
- I have tried browsing to the meshes folder and adding a file but it seems
that I can only add code.
- I have tried using Github desktop but it seems like it will start a new
pull request when I add the file to my local folder and hit sync.

Please let me know.  Thanks for your understanding and patience with me.


*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Wed, Sep 9, 2015 at 2:06 AM, Tim Kuipers <notifications@github.com>
wrote:

> It's best if you add the stl in this pull request.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-138808972>.
>

BagelOrb commented:
Ok whatever, do it in a new pull request. It doesn't matter that much.
I will then merge the two at the same time.

adamarumjahn commented:
Done.  I just added a commit to add the STL file as well.  Thanks for your
patience.  Still learning how to use Github but getting better and better
every day.  Let me know if I did everything right.  If there's anything you
need me to fix please let me know.

Thanks.

*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Thu, Sep 10, 2015 at 2:29 AM, Tim Kuipers <notifications@github.com>
wrote:

> Ok whatever, do it in a new pull request. It doesn't matter that much.
> I will then merge the two at the same time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-139138317>.
>

awhiemstra commented on resources/settings/innovo-inventor.json:
```diff
@@ -0,0 +1,48 @@
+        "wall_thickness": { "default": 0.8 },
+        "top_bottom_thickness": { "default": 0.3, "visible": true },
+        "material_print_temperature": { "default": 215, "visible": true },
+        "material_bed_temperature": { "default": 60, "visible": true },
```
You're setting Bed Temperature to visible but not setting the machine setting "machine_heated_bed" to true. This will mean you will not get the right heated bed behaviour since several parts of the code check the machine_heated_bed setting.

adamarumjahn replied:
You're right.  I wasn't intending to have the heated bed installed but was
asked to make it visible even if it was not used.  However, I now plan to
have the heated bed installed. So I will change this right away.

*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Sun, Sep 13, 2015 at 2:26 PM, awhiemstra <notifications@github.com>
wrote:

> In resources/settings/innovo-inventor.json
> <https://github.com/Ultimaker/Cura/pull/301#discussion_r39350957>:
>
> > +        "machine_head_shape_max_y": {"default": 55},
> > +        "machine_nozzle_gantry_distance": {"default": 82.3},
> > +        "machine_nozzle_offset_x_1": {"default": 0},
> > +        "machine_nozzle_offset_y_1": {"default": 15},
> > +        "machine_gcode_flavor": {"default": "RepRap (Marlin/Sprinter)"},
> > +        "machine_start_gcode": {"default": "G28 ; Home extruder\nM107 ; Turn off fan\nG90 ; Absolute positioning\nM82 ; Extruder in absolute mode\n{IF_BED}M190 S{BED}\n{IF_EXT0}M104 T0 S{TEMP0}\n{IF_EXT0}M109 T0 S{TEMP0}\n{IF_EXT1}M104 T1 S{TEMP1}\n{IF_EXT1}M109 T1 S{TEMP1}\nG32 S3 ; auto level\nG92 E0 ; Reset extruder position"},
> > +        "machine_end_gcode": {"default": "M104 S0\nG91 ; relative positioning\nG1 E-2 F5000; retract 2mm\nG28 Z; move bed down\nG90 ; absolute positioning\nM84   ; disable motors"}
> > +    },
> > +
> > +    "overrides": {
> > +        "layer_height": { "default": 0.15 },
> > +        "shell_thickness": { "default": 0.8},
> > +        "wall_thickness": { "default": 0.8 },
> > +        "top_bottom_thickness": { "default": 0.3, "visible": true },
> > +        "material_print_temperature": { "default": 215, "visible": true },
> > +        "material_bed_temperature": { "default": 60, "visible": true },
>
> You're setting Bed Temperature to visible but not setting the machine
> setting "machine_heated_bed" to true. This will mean you will not get the
> right heated bed behaviour since several parts of the code check the
> machine_heated_bed setting.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301/files#r39350957>.
>

adamarumjahn replied:
Hey everyone, I've been checking other JSON files and I don't see anyone else setting "machine_heated_bed" to true, not even in the Ultimaker JSON file.  Am I missing something?  Does this value have to be set to true?

awhiemstra replied:
See https://github.com/Ultimaker/Cura/blob/master/resources/machines/ultimaker2.json#L39 for an example or https://github.com/Ultimaker/Cura/blob/master/resources/machines/ultimaker_original_plus.json for an example that does little more than set heated bed.

For the original, since it does not be default have a heated bed, the heated bed option is set from the upgrades page in the add machine wizard. So that is a more complicated case. :)

awhiemstra commented:
@adamarumjahn Any update on this?

adamarumjahn commented:
I totally forgot about this.  Thanks for reminding me.  Just updated it.

*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Fri, Sep 25, 2015 at 4:04 AM, awhiemstra <notifications@github.com>
wrote:

> @adamarumjahn <https://github.com/adamarumjahn> Any update on this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-143165447>.
>

awhiemstra commented:
It looks like the build platform file is still missing from this request but I can see it is available at https://github.com/innovo3d/Cura/tree/master/resources/meshes . If you want I can merge this PR and copy the STL over manually.

innovo3d commented:
Please do. Thanks!
On Sep 28, 2015 4:48 AM, "awhiemstra" <notifications@github.com> wrote:

> It looks like the build platform file is still missing from this request
> but I can see it is available at
> https://github.com/innovo3d/Cura/tree/master/resources/meshes . If you
> want I can merge this PR and copy the STL over manually.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-143695059>.
>

BagelOrb commented:
@awhiemstra Could you look into this then?

awhiemstra commented:
Yes sorry, lots of stuff on my mind at the moment.

innovo3d commented:
Hey guys.  I'm reviewing the latest release that includes this machine profile.  Looks awesome by the way, great job guys!

1. Can I still update this json file?  I tried committing to this Pull request from Github desktop but it appears that it may not update since the PR has already been merged and closed.  (please forgive my ignorance, I'm a noob to Github).  Should I open a new PR?

2. I have one additional question.  I use auto levelling with Repetier firmware; G32 S3 which auto levels every time before a print.  I'm planning on changing this so that it DOESN'T auto-level before every print, rather, I auto level once and save the transformation matrix... the gcode generated for every print will simply print without auto level in order to save time.

Since Cura doesn't allow sending g-codes directly to the printer, is there another way that we can send Auto-level commands from the interface?  Let me know if there is a workaround.

fieldOfView commented:
> Looks awesome by the way, great job guys!
If you're talking about 2.1, you ain't seen nothing yet ;-)

If you have any improvements on the definition, please push them against master. You will see that the file has changed a bit. A new PR would be best.

You're right, we currently don't have a way to send custom gcode to the printer "on demand". In 2.2 we have a much improved (USB-)print-monitor. It would be nice if we could allow plugins in it to send custom gcode.

innovo3d commented:
Thanks Aldo.  I haven't looked into the plug-ins/scripts.  By chance, is
there a way to use those for sending auto-level gcodes in the meantime?

*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Tue, Aug 16, 2016 at 1:35 AM, Aldo Hoeben <notifications@github.com>
wrote:

> Looks awesome by the way, great job guys!
> If you're talking about 2.1, you ain't seen nothing yet ;-)
>
> If you have any improvements on the definition, please push them against
> master. You will see that the file has changed a bit. A new PR would be
> best.
>
> You're right, we currently don't have a way to send custom gcode to the
> printer "on demand". In 2.2 we have a much improved (USB-)print-monitor. It
> would be nice if we could allow plugins in it to send custom gcode.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-240015526>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHIXa7QfOZfQVnAwR6pjEOAvkCwrzv2Cks5qgVpEgaJpZM4Fy3er>
> .
>

fieldOfView commented:
> is there a way to use those for sending auto-level gcodes in the meantime?

No, not without a lot of work. It should not be too long before there is a beta of 2.2.

innovo3d commented:
Ok thanks Aldo.  In the meantime, I've updated 2 commits to branch 2.1.
 (commit ID number 312dbbf).  Are these automatically updated into 2.1 or
do I have to send a pull request for these?  I tried a Pull Request, but it
looks like it tries to merge it into Master.  Once again, forgive me, I'm a
noob with Github still.

As for 2.2, I see that the profiles are now in a Definitions folder and the
version is now set to 2.  So I'll start by modifying the new files that you
have created and update them with the same changes.  Right?

Thanks,


*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Tue, Aug 16, 2016 at 11:08 AM, Aldo Hoeben <notifications@github.com>
wrote:

> is there a way to use those for sending auto-level gcodes in the meantime?
>
> No, not without a lot of work. It should not be too long before there is a
> beta of 2.2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-240151389>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHIXa2llkUWXbtpzJtRXN9DZZwvzfhHZks5qgeCDgaJpZM4Fy3er>
> .
>

fieldOfView commented:
2.1 is feature-frozen, so chances your changes will be included into 2.1 are slim to none. If we were still accepting changes for 2.1, we would need a new PR. That is also the way to go for 2.2/master.

innovo3d commented:
I just tried the PR for 2.1 and I got this:

The command is disabled and cannot be executed.
[image: Inline image 1]

Any ideas?  I'm attaching the files just in case.



*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Tue, Aug 16, 2016 at 4:14 PM, Aldo Hoeben <notifications@github.com>
wrote:

> 2.1 is feature-frozen, so chances your changes will be included into 2.1
> are slim to none. If we were still accepting changes for 2.1, we would need
> a new PR. That is also the way to go for 2.2/master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-240241000>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHIXa4hs0-kuPuva_5gtBYOxgkyvvV9Uks5qgigagaJpZM4Fy3er>
> .
>

Ghostkeeper commented:
Your image doesn't work, but searching for that error message online lands me on a bug report for Eclipse.

My common approach to creating pull requests is to use Github Desktop (which makes it very easy) or if I'm on Linux, use the "Create Pull Request" button on the project's main page on github.com.

innovo3d commented:
I was using Github Desktop and thats where the error appeared.  I'm going
to try reinstalling Github again to see if the problem goes away.  I have a
feeling it has to do with the SSH keys, but I've since renewed them and it
still doesn't seem to be working.

Noob question.  What should my workflow be like?

This is how I'm doing it.  Go to Cura/Ultimaker and Clone to Github
Desktop.  For changes to 2.1, I sync the branch and then start
changing/adding my files.  Then I Commit the changes and send a Pull
Request.  Is this correct?

Or should I be forking the Cura/Ultimaker repository into my own directory
/Innovo3d/Cura and make the Commits from there?

Perhaps that might answer the question about permissions and access..

Thanks a lot in advance.

*Adam Rumjahn*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Tue, Aug 16, 2016 at 6:04 PM, Ghostkeeper <notifications@github.com>
wrote:

> Your image doesn't work, but searching for that error message online lands
> me on a bug report for Eclipse.
>
> My common approach to creating pull requests is to use Github Desktop
> (which makes it very easy) or if I'm on Linux, use the "Create Pull
> Request" button on the project's main page on github.com.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-240266386>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHIXa88klpacJWa25WVq4yaAaphe49V-ks5qgkH5gaJpZM4Fy3er>
> .
>

Ghostkeeper commented:
Yes, I've always worked from a fork. Perhaps that was your problem.

innovo3d commented:
Yes, that seems to have been my problem.  I deleted everything and forked again from Master and have finally submitted my changes with a new PR for 2.1.  Thanks for all of your help and patience everyone.  Will these be updated anytime soon?  I'm launching a Makerspace with these printers soon and it would be nice if the JSON files were updated in the install image so I don't have to go modifying everyone's installation manually.

Also, I wanted to start making changes to the def.JSON files in the Master branch, but when I switched back to Master, it says I have 37 uncommitted changes???  Looking at these files, I never modified any of these!!  

Can someone please guide me?  I just want to modify my JSON file.  Thanks.
<img width="1552" alt="screen shot 2016-08-17 at 4 09 11 pm" src="https://cloud.githubusercontent.com/assets/7477099/17753323/fdd7da96-6494-11e6-9f52-8dd6bde2e223.png">

Ghostkeeper commented:
Haha, you seem to have some merge conflicts now. They don't make it easy, do they. By the looks of it, something still went wrong with the 2.1 branch.

If you want an easy way to cop out, you could send the def.json file over via e-mail. My address rubend at tutanota dot com (as can be seen on my commits if you type "git log" in a terminal from Cura's source repository). I'll check your file and have it added in your name. I can even commit it as you.

Otherwise, the least technical way to solve this would probably be to make a back-up of your .def.json file, delete your whole fork, and make a new one. Then put your .def.json file back in.

A more technical way would be to make a back-up of your .def.json file, open a terminal in Cura's source directory (via Github Desktop, click on the gear in the top right and select "Open in Git Shell"), and type `git checkout .` there to erase all uncommitted changes. Then try `git pull origin master` to make sure you're up to date (`git status` should display no changes). Put your .def.json file back and try again.

innovo3d commented:
@Ghostkeeper, thanks so much for the help and offer.  I ended up removing the repo, deleting the files and cloning the forked repo again.  All is good now.  Again, thanks everyone here for your patience.  I'm starting to get the hang of Github/Desktop and the workflow now.

I've submitted PRs for 2.1 and 2.2/Master.  I'm hoping that 2.1 will still get updated soon and it might be in your best interest for keeping the image as light as possible.  For some reason my machine STL file was 13MB.  I thought I successfully uploaded the compressed version last year but apparently I didn't.  The new version in the pull request is 810kb.

Ghostkeeper commented:
We've done the last update for 2.1 that was planned, last week, which was 2.1.3.

The age of 2.1 is over. The time of 2.2 is to come.

awhiemstra commented:
> Since Cura doesn't allow sending g-codes directly to the printer, is there another way that we can send Auto-level commands from the interface? Let me know if there is a workaround.

Note that for 2.3 you could create a custom machine action that sends this command to the printer. Probably not the easiest solution but it would be the most elegant. :)

innovo3d commented:
Hi guys, I´ve been updating my definition and tried to comment G32 out for
the latest release but for some reason it wasn´t pushed.  What am I doing
wrong?




<https://mailtrack.io/> Sent with Mailtrack
<https://chrome.google.com/webstore/detail/mailtrack-for-gmail-inbox/ndnaehgpjlnokgebbaldlmgkapkpjkkb?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality>

*Adam Rumjahn*
*CTO & Founder*

* <http://www.innovo3d.co>*

*adam@innovo3d.co <adam@innovo3d.co>Cel: +57 304 643 1248*
*Fijo: +57 2 374 2473*

On Mon, Sep 5, 2016 at 6:29 AM, awhiemstra <notifications@github.com> wrote:

> Since Cura doesn't allow sending g-codes directly to the printer, is there
> another way that we can send Auto-level commands from the interface? Let me
> know if there is a workaround.
>
> Note that for 2.3 you could create a custom machine action that sends this
> command to the printer. Probably not the easiest solution but it would be
> the most elegant. :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Ultimaker/Cura/pull/301#issuecomment-244724980>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHIXa9ylJzTkwUs6tBsTf4vJalSF6Lswks5qm_0LgaJpZM4Fy3er>
> .
>

Ghostkeeper commented:
Looks like you found the problem. There's a new pull request here: https://github.com/Ultimaker/Cura/pull/2914
